src/axom/primal/geometry/Sphere.hpp

```diff
@@ -6,14 +6,16 @@
 #ifndef AXOM_PRIMAL_SPHERE_HPP_
 #define AXOM_PRIMAL_SPHERE_HPP_
 
```

Comment:
The changes to `primal::Sphere` change its API and implementation to be based on primal primitives -- e.g. it is based on `primal::Point` (which always has the right dimensions) instead of a pointer to an array.